### PR TITLE
ees-5843 Dynamic boundary footnote bugfix & Maps across time robot test

### PR DESCRIPTION
**Pull request by bennettstuart**
Fixes maps across time to allow for the boundary footnote to update when the boundaries are changed in the map.
Also added a robot test to check both map boundaries and footnotes change when selecting data sets with different boundaries

### CODE REVIEW

**Review by amyb-hiveit — APPROVED** [review]
A couple of minor queries but otherwise looks good 👍

File: src/explore-education-statistics-common/src/modules/charts/components/ChartRenderer.tsx [hunk 1]
```diff
@@ -61,33 +65,52 @@ function ChartRenderer({ source, id, chart }: ChartRendererProps) {
...
         return <InfographicBlock {...chart} />;
       default:
         return <p>Unable to render invalid chart type</p>;
     }
   }, [id, chart]);
 
-  if (data?.length > 0 && meta) {
-    const footnotes = [...meta.footnotes];
+  const footnotes = useMemo(() => {
+    if (!(data?.length > 0) || !meta) {
```
[the remark this answers is not in the document]
Comment (bennettstuart) [reply src/explore-education-statistics-common/src/modules/charts/components/ChartRenderer.tsx]: Great point! fix that, ta

File: src/explore-education-statistics-common/src/modules/charts/components/ChartRenderer.tsx [hunk 2]
```diff
@@ -61,33 +65,52 @@ function ChartRenderer({ source, id, chart }: ChartRendererProps) {
...
+      return [];
+    }
 
+    const metaFootnotes = [...meta.footnotes];
     const boundaryFootnoteId = 'map-footnote';
+
     if (
-      type === 'map' &&
-      footnotes.findIndex(footnote => footnote.id === boundaryFootnoteId) === -1
+      chart.type === 'map' &&
```
[the remark this answers is not in the document]
Comment (bennettstuart) [reply src/explore-education-statistics-common/src/modules/charts/components/ChartRenderer.tsx]: Great point! fix that, ta
